authentication/auth.go

```diff
@@ -522,6 +529,28 @@ func GenerateTokenFromDefaultsV2(schemaURL string, accountServiceURL string, url
 	return token, ""
 }
 
+func addUrlBooleanMetadata(updatedMetadata map[string]interface{}, requiredSchemaMetadata []Metadata) []Metadata {
+	for metadataName, metadataValue := range updatedMetadata {
+		if !(isExistingRequiredSchemaMetadata(requiredSchemaMetadata, metadataName)) {
```

Comment:
I think this if statement can be removed which will reduce the complexity here, it looks like the duplicate value that would be generated is already being handled and overwritten elsewhere. The token generated looks fine in runner without this 👍

Reply:
Makes sense, updated 👍